Automattic/newspack-blocks
# Donate Block: Accessibility improvements

### All Submissions:


### Changes proposed in this Pull Request:

This PR makes accessibility improvements to the Frequency layout version of the Donate block, specifically to make sure the tabs have the correct markup and attributes, and can be navigated in a predictable way via keyboard.

The tabs have been added on top of the existing radio button logic that the block used, to try to minimize the changes being made. 

See 1200550061930446-as-1203715106299306 and 1200024937525366-as-1202002498910398

### How to test the changes in this Pull Request: 

1. Apply this PR and run `npm run build`.
2. Add a donate block to the editor, and switch it to the 'Frequency' Layout. 
3. In the editor, try clicking the different tabs, and confirm that the active tab and panel updates. 
4. In the editor, try switching the 'default' tab to one that's neither the original default nor the currently selected one, and confirm that the correct tab becomes selected. Repeat step 3 to confirm the other tabs still change on click.
5. Enable Configure Manually, and try disabling one tab; confirm that the preview works, and that setting the Default tab or clicking on the tabs still works. Re-enable the tab so all are visible. 
6. Publish the block. 
7. Repeat steps 3-5 on the front-end, and confirm that the block works as expected: the tabs and content update on click, and they display as expected. 
8. Try changing the block styles to Alternate and Minimal, and confirm that things display as expected. 

Default:
![image](https://github.com/Automattic/newspack-blocks/assets/177561/7e06705b-ccb7-487d-ab23-2d04b594493d)

Alternate:
![image](https://github.com/Automattic/newspack-blocks/assets/177561/8a66fce3-5b49-4245-a820-f1ddafb31de2)

Minimal:
![image](https://github.com/Automattic/newspack-blocks/assets/177561/46f5296e-f5bb-4023-aa90-ec1790492638)

9. It would also be good to do a visual test of the Tiered Donate block to make sure nothing has changed -- the two blocks don't share the same markup, but they do share some of the same styles, so the changes made in this PR may affect that block.

These are the specific accessibility improvements that were added in this update:

* You can now use the "tab" key to navigate between different frequencies.
* The tabs have been replaced with buttons (which is required for them to to be "tabable" with the keyboard, and is semantically correct for their function). Each of these buttons has the attribute `role="tab"`, and all three are wrapped in a parent element with the attribute `role="tablist"`.
* Each tab and its associated panel has attributes that link them together -- each tab has a unique ID and an `aria-controls` attribute that has the ID of its associated panel, and each panel has an ID and an `aria-labelledby` attribute that has the ID of its associated tab.  
* When a tab is selected, it will have the attribute `aria-selected="true"`; if not, it will have the attribute `aria-selected="false"`
* When a tab is selected, its associated panel should get the attribute `tabindex="0"`. 

### Other information:
